Portfolio/src/main/java/com/google/sps/servlets/GetCommentsServlet.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+
+@WebServlet("/get_comments")
+public class GetCommentsServlet extends HttpServlet {
```

Comment:
I like that you want to make these very coherent and pull them into two separate classes, however I think there's a minimal amount of logic and it actually makes sense to put the GetComments and CreateComments actions in the same file (see my other comment for how we could get benefits here).

I think we should move the `getComments` and `doPost` methods into the common `CommentServlet`. doPost here should be renamed to `doGet` as GET requests return data from the server while POST/PUT requests entail the user will be writing data to be persisted on the server.

Reply:
I have not many experience with backend, so this comments are great feedback for me.
I knew that GET, POST, PUT, entail different actions but forgot to apply it here.

Also again, this change made the code a lot more compact, readable and concise, good design idea.